crates/sui-framework/sources/governance/validator_cap.move

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::validator_cap {
+    use sui::object::{Self, ID, UID};
+    use sui::transfer;
+    use sui::tx_context::TxContext;
+    friend sui::sui_system;
+    friend sui::validator;
+    friend sui::validator_set;
+
+    #[test_only]
+    friend sui::sui_system_tests;
+    #[test_only]
+    friend sui::rewards_distribution_tests;
+
+    /// The capability object is created when creating a new `Validator` or when the
+    /// validator explicitly creates a new capability object for rotation/revocation.
+    /// The holder address of this object can perform some validator operations on behalf of
+    /// the authorizer validator. Thus, if a validator wants to separate the keys for operation
+    /// (such as reference gas price setting or tallying rule reporting) from fund/staking, it
+    /// could transfer this capability object to another address.
+
+    /// To facilitate rotating/revocation, `Validator` stores the ID of currently valid
+    /// `UnverifiedValidatorOperationCap`. Thus, before converting `UnverifiedValidatorOperationCap`
+    /// to `ValidatorOperationCap`, verification needs to be done to make sure
+    /// the cap object is still valid.
+    struct UnverifiedValidatorOperationCap has key, store {
+        id: UID,
+        authorizer_validator_address: address,
+    }
+
+    /// Privileged operations require `ValidatorOperationCap` for permission check.
+    /// This is a Hot Potato, only constructed after successful verification,
+    /// and deconstructed after use.
+    struct ValidatorOperationCap {
+        authorizer_validator_address: address,
+    }
+
+    public(friend) fun unverified_operation_cap_address(cap: &UnverifiedValidatorOperationCap): &address {
+        &cap.authorizer_validator_address
+    }
+
+    public(friend) fun verified_operation_cap_address(cap: &ValidatorOperationCap): &address {
+        &cap.authorizer_validator_address
+    }
+
+    /// Should be only called by the friend modules when adding a `Validator`
+    /// or rotating an existing validaotr's `operation_cap_id`.
+    public(friend) fun new_unverified_validator_operation_cap_and_transfer(
```

Comment:
Do we want to call it `rotate...` instead of `new...` to make it clear that the old cap is now revoked and invalid?

Reply:
resolved offline and updated